### PR TITLE
Add dbt-materialize adapter docs

### PR DESCRIPTION
**Pull request by jtcohen6**
cc @JLDLaughlin

## Description & motivation
- Add docs for [`dbt-materialize`](https://github.com/MaterializeInc/materialize/blob/main/misc/dbt-materialize/README.md) adapter. I copy-pasted the notes about Materializations and Seeds, which are wonderful; let me know if you feel strongly about excluding these here, and linking out to the other docs instead, so as to avoid stagnation.
- Add `dbt-materialize` to list of Available Adapters under a new "Vendor Supported" section. Open to feedback on the language here.

## Pre-release docs
Is this change related to an unreleased version of dbt?
- [ ] Yes: please
    - update the base branch to `next`
    - add Changelog components: `<Changelog>[New/Changed] in v0.x.0</Changelog>`
    - add links to the "New and changed documentation" section of the latest [Migration Guide](../website/docs/docs/guides/migration-guide)
- [x] No: please ensure the base branch is `current`
- [ ] Unsure: we'll let you know!

## Checklist
If you added new pages (delete if not applicable):
- [x] The page has been added to `website/sidebars.js`
- [x] The new page has a unique filename

### CODE REVIEW

File: website/docs/docs/available-adapters.md
Comment (amychen1776) [inline website/docs/docs/available-adapters.md:30]: This is more of my own curiosity but is there a reason why this lives in a misc folder rather than a separate repo from the Materialize product repo?

File: website/docs/reference/warehouse-profiles/materialize-profile.md
Comment (amychen1776) [inline website/docs/reference/warehouse-profiles/materialize-profile.md]: I think you meant Materialize here 😁

File: website/docs/reference/warehouse-profiles/materialize-profile.md
Comment (amychen1776) [inline website/docs/reference/warehouse-profiles/materialize-profile.md]: Could we say something more along the lines of go to the linked repository + then open an issue for the maintainers? See the source code sounds off to me.

File: website/docs/reference/warehouse-profiles/materialize-profile.md
Comment (amychen1776) [inline website/docs/reference/warehouse-profiles/materialize-profile.md]: > Better yet, our materialized views will always

This is super nitpicky 😓 But the word `our` seems off because it's on the dbt docs. I think it makes sense for dropping `our` when it comes to speaking for Materialize and speak in the more neutral like `the documentation` or just `materialized views`

File: website/docs/reference/warehouse-profiles/materialize-profile.md
Comment (amychen1776) [inline website/docs/reference/warehouse-profiles/materialize-profile.md]: Does this mean that `dbt seed --full-refresh` will not update the view? Should we say the view needs to be dropped in order to update it?

File: website/docs/reference/warehouse-profiles/materialize-profile.md
Comment (clrcrl) [inline website/docs/reference/warehouse-profiles/materialize-profile.md]:
```suggestion
The dbt profile for Materialize is nearly identical to the [profile configuration for Postgres](profile-postgres):
```
```suggestion
The dbt profile for Materialize is nearly identical to the [profile configuration for Postgres](profile-postgres):
```
```suggestion
The dbt profile for Materialize is nearly identical to the [profile configuration for Postgres](postgres-profile):
```

File: website/docs/reference/warehouse-profiles/materialize-profile.md
Comment (amychen1776) [inline website/docs/reference/warehouse-profiles/materialize-profile.md]: Nitpicky again 😓  Interactive implies something more like the Materialize demo. I think this is actually just more of a practice tutorial.

**Review by amychen1776 — CHANGES_REQUESTED** [review]
Thank you for writing this up! I'm so excited to see this live.

File: website/docs/reference/warehouse-profiles/materialize-profile.md
Comment (clrcrl) [inline website/docs/reference/warehouse-profiles/materialize-profile.md]: Let's adjust this to be less "our" in case people think that Fishtown Analytics are talking about this

File: website/docs/docs/available-adapters.md
Comment (clrcrl) [inline website/docs/docs/available-adapters.md:24]: I think this is the right terminolgy!

**Review by clrcrl — COMMENTED** [review]
ha, I think Amy and I were looking at this at the same time. I'll hold off on a deeper review since I think Amy caught most of the things I was going to catch! (And probably caught a few extra ones)

File: website/docs/docs/available-adapters.md
Comment (JLDLaughlin) [reply website/docs/docs/available-adapters.md:30]: Keeping it in our main repository (under `misc`) allows us to easily run the `dbt-adapter-tests` as part of our CI workflow!

Comment (amychen1776) [reply website/docs/docs/available-adapters.md:30]: ah! That's good to know! Thanks @JLDLaughlin

File: website/docs/reference/warehouse-profiles/materialize-profile.md
Comment (jtcohen6) [reply website/docs/reference/warehouse-profiles/materialize-profile.md]: Correct, just confirmed this, good call

File: website/docs/reference/warehouse-profiles/materialize-profile.md
Comment (jtcohen6) [reply website/docs/reference/warehouse-profiles/materialize-profile.md]: I'll call this "Get Started" instead. This is nearly identical to Materialize's own [Get Started demo](https://materialize.com/docs/get-started/), just using dbt instead of manually running all the DDL

**Comment by jtcohen6** [comment]
Thanks for taking a look @amychen1776 @clrcrl @JLDLaughlin! (And thanks for fixing the deploy preview, Claire!)

If one of you could give it another glance later today or tomorrow, I think we'll be good to go (:

File: website/docs/reference/warehouse-profiles/materialize-profile.md
Comment (amychen1776) [inline website/docs/reference/warehouse-profiles/materialize-profile.md]:
```suggestion
Certain core functionality may vary. If you would like to report a bug, request a feature, or contribute, you can check out the linked repository and open an issue.
```

**Review by amychen1776 — APPROVED** [review]
One minor suggestion but this looks great :)

**Pull request merged**
